src/ngx_http_set_hashed_upstream.c

```diff
@@ -25,32 +35,73 @@ ngx_http_set_misc_apply_distribution(ngx_log_t *log, ngx_uint_t hash,
 }
 
 
+uint32_t
+ngx_http_set_misc_bsearch(
+    ngx_http_set_hashed_upstream_consistent_hash_node *nodes, uint32_t node_len,
+    uint32_t target)
+{
+    uint32_t m = 0, n = node_len, l;
+
+    while (m < n) {
+        l = (m + n) >> 1;
+        if (target == nodes[l].hash) {
+            return l;
+        }
+        if (target < nodes[l].hash) {
+            n = l;
+        } else {
+            m = l + 1;
+        }
+    }
+    return m;
+}
+
+
+int
+ngx_http_set_misc_consistent_hash_compare(
+    const ngx_http_set_hashed_upstream_consistent_hash_node *node1,
+    const ngx_http_set_hashed_upstream_consistent_hash_node *node2)
+{
+    if (node1->hash < node2->hash) {
+        return -1;
+
+    } else if (node1->hash > node2->hash) {
```

Comment:
Because the previous branches always return, we can avoid the `else if` part here.